modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java

```diff
@@ -268,4 +271,19 @@ public void testNotFoundErrorMessageContainsFullKey() throws Exception {
             assertThat(exception.getMessage(), containsString("nested/dir/blob] not found"));
         }
     }
+
+    public void testReadByteByByte() throws Exception {
+        try (BlobStore store = newBlobStore()) {
+            BlobContainer container = store.blobContainer(BlobPath.EMPTY.add(UUIDs.randomBase64UUID()));
+            var data = randomBytes(randomIntBetween(128, 512));
+            String blobName = randomName();
+            container.writeBlob(blobName, new ByteArrayInputStream(data), data.length, true);
+            try (var inputStream = container.readBlob(blobName)) {
+                for (byte blobByte : data) {
+                    assertThat(blobByte, is(equalTo((byte) inputStream.read())));
```

Comment:
One detail here: can we expand `blobByte` instead of narrowing the result of `inputStream.read()`? That would validate the `& 0xff` works correctly.

```suggestion
                    assertThat((int) blobByte, is(equalTo(inputStream.read())));
```